js/views/modals/orderDetail/ResolveDispute.js

```diff
@@ -0,0 +1,173 @@
+import $ from 'jquery';
+import { getAvatarBgImage } from '../../../utils/responsive';
+import {
+  resolvingDispute,
+  resolveDispute,
+  events as orderEvents,
+} from '../../../utils/order';
+import loadTemplate from '../../../utils/loadTemplate';
+import BaseVw from '../../baseVw';
+
+export default class extends BaseVw {
+  constructor(options = {}) {
+    super(options);
+
+    if (!this.model) {
+      throw new Error('Please provide an OrderFulfillment model.');
+    }
+
+    const isValidParticipantObject = (participant) => {
+      let isValid = true;
+      if (!participant.id) isValid = false;
+      if (typeof participant.getProfile !== 'function') isValid = false;
+      return isValid;
+    };
+
+    const getInvalidParticpantError = (type = '') =>
+      (`The ${type} object is not valid. It should have an id ` +
+        'as well as a getProfile function that returns a promise that ' +
+        'resolves with a profile model.');
+
+    if (!options.vendor) {
+      throw new Error('Please provide a vendor object.');
+    }
+
+    if (!isValidParticipantObject(options.vendor)) {
+      throw new Error(getInvalidParticpantError('vendor'));
+    }
+
+    if (!options.buyer) {
+      throw new Error('Please provide a buyer object.');
+    }
+
+    if (!isValidParticipantObject(options.buyer)) {
+      throw new Error(getInvalidParticpantError('buyer'));
+    }
+
+    options.buyer.getProfile().done(profile => {
```

Comment:
It's not necessary because all the profile are lazy loaded in and the view supports them not being there. For example, let's say we display the text "Billy made a purchase" with Billy being the buyer's name. Until Billy's profile is loaded, it will say "The buyer made a purchase" and when the profile arrives the name will pop in. When it's cached, it's so fast you barely notice the change.

In every case it should work similarly (sometimes, it's a GUID in place of handle, etc...).